### PR TITLE
manage spirv_arch & physical_addressing in device/device_properties

### PR DESCRIPTION
**Pull request by rjodinchr**
This will help once we have something detected whether or not to use physical addressing.
It will also help force some device to not use physical addressing if the feature is bugged.

### CODE REVIEW

**Comment by rjodinchr** [comment]
We need to rebuild swiftshader. It seems the artifact has expired thus removed.

File: src/device.hpp [hunk 1]
```diff
@@ -528,6 +528,9 @@ struct cvk_device : public _cl_device_id,
         return m_max_first_cmd_group_size;
     }
 
+    const std::string get_spirv_arch() const { return m_spirv_arch; }
```
Comment (kpet) [inline src/device.hpp:531]:
```suggestion
```

No longer needed if we introduce `address_bits`.

File: src/device.hpp [hunk 2]
```diff
@@ -528,6 +528,9 @@ struct cvk_device : public _cl_device_id,
...
+    bool get_physical_addressing() const { return m_physical_addressing; }
```
Comment (kpet) [inline src/device.hpp:532]:
```suggestion
    bool uses_physical_addressing() const { return m_physical_addressing; }
```

**Review by kpet — CHANGES_REQUESTED** [review]
Agree this is a needed change and good direction. Just a few minor changes please.

File: src/api.cpp [hunk 3]
```diff
@@ -529,7 +529,7 @@ cl_int CLVK_API_CALL clGetDeviceInfo(cl_device_id dev,
         size_ret = sizeof(val_fpconfig);
         break;
     case CL_DEVICE_ADDRESS_BITS:
-        val_uint = (config.spirv_arch() == "spir64") ? 64 : 32;
+        val_uint = (device->get_spirv_arch() == "spir64") ? 64 : 32;
```
[the remark this answers is not in the document]
Comment (rjodinchr) [reply src/api.cpp]: Done

File: src/memory.hpp [hunk 4]
```diff
@@ -62,6 +64,7 @@ struct cvk_memory_allocation {
     VkDeviceSize m_size;
     VkDeviceMemory m_memory;
     uint32_t m_memory_type_index;
+    bool m_physical_addressing;
```
[the remark this answers is not in the document]
Comment (rjodinchr) [reply src/memory.hpp]: Done

File: src/device.hpp (hunk 1 above)
Comment (rjodinchr) [reply src/device.hpp]: Done

File: src/device.hpp (hunk 2 above)
Comment (rjodinchr) [reply src/device.hpp]: Done

**Review by kpet — APPROVED** [review]
Thanks for the changes!

**Pull request merged**